### PR TITLE
[bitnami/mongodb] Added the possibilty to define number of replicas in standalone mode

### PR DESCRIPTION
**Pull request by rubij**
### Description of the change

Currently, standalone mode always spawn 1 replicas of Mongodb : that setting could be parametrized from the values file.
### Benefits

Allow the possibility to not start Mongodb by setting values.replicaCount to 0. Useful in case we want to do any upgrade / modification but not start the pod when doing a Helm upgrade command.

### Possible drawbacks

N/A

### Applicable issues

N/A

### Additional information

N/A
### Checklist
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by FraPazGal — COMMENTED** [review]
Hello @rubij! If I'm understanding your use case correctly, you want the option to upgrade a standalone setup while maintaining the number of replicas at zero right? is using `kubectl scale --replicas=0` a viable workaround for you?

My main concern with your current proposal is modifying the default value of `replicaCount`, which would affect users using the replicaset architecture. If parametrising the number of replicas for the standalone template is the only solution we'll need to add logic so current replicaset users are not affected.

**Comment by rubij** [comment]
Hi @FraPazGal, unfortunately using `kubectl` to scale the replicas to 0 will not be an option for us.

I could revert the default value of `replicaCount` to 2 to not affect users using the replicaSet architecture.

What logic do you propose for the standalone template then ?

**Comment by rubij** [comment]
> Hi @FraPazGal, unfortunately using `kubectl` to scale the replicas to 0 will not be an option for us.
> 
> I could revert the default value of `replicaCount` to 2 to not affect users using the replicaSet architecture.
> 
> What logic do you propose for the standalone template then ?

Hi @FraPazGal, unfortunately using kubectl to scale the replicas to 0 will not be an option for us.

I could revert the default value of replicaCount to 2 to not affect users using the replicaSet architecture.

What logic do you propose for the standalone template then ?

**Comment by rubij** [comment]
@FraPazGal may I have an update on this ?

**Comment by FraPazGal** [comment]
Hello @rubij, apologies about the late response. The ideal solution would be to avoid modifying any of the chart's current behaviour. We need to prevent an user upgrading after this PR and encountering 1 of the following cases:
- If using `standalone` and we leave `replicaCount=2` by default, the user will find its chart scaling up `1->2`.
- If using `replicaset`  and we change the default to `replicaCount=1`, we would break the chart.

Another concern is whether having a standalone setup with `>1` replicas makes sense. I would need to run some tests but I'll say the MongoDB wouldn't even support that configuration (we have the replicaset config and bitnami/mongodb-sharded chart for that). Although messy, a compromise would be to only allow the standalone template to have zero or one replica.

Taking all of the above into account, what do you think of leaving the default as `replicaCount=2` and making it so the [standalone template](https://github.com/bitnami/charts/blob/main/bitnami/mongodb/templates/standalone/dep-sts.yaml) only uses `.Values.replicaCount` when it has a value of `0`?

**Comment by rubij** [comment]
> Hello @rubij, apologies about the late response. The ideal solution would be to avoid modifying any of the chart's current behaviour. We need to prevent an user upgrading after this PR and encountering 1 of the following cases:
> 
> * If using `standalone` and we leave `replicaCount=2` by default, the user will find its chart scaling up `1->2`.
> * If using `replicaset`  and we change the default to `replicaCount=1`, we would break the chart.
> 
> Another concern is whether having a standalone setup with `>1` replicas makes sense. I would need to run some tests but I'll say the MongoDB wouldn't even support that configuration (we have the replicaset config and bitnami/mongodb-sharded chart for that). Although messy, a compromise would be to only allow the standalone template to have zero or one replica.
> 
> Taking all of the above into account, what do you think of leaving the default as `replicaCount=2` and making it so the [standalone template](https://github.com/bitnami/charts/blob/main/bitnami/mongodb/templates/standalone/dep-sts.yaml) only uses `.Values.replicaCount` when it has a value of `0`?

@FraPazGal I understand the concerns you mentioned, so:

- we could leave the replicaCount=2 default value (replicaSet)
- we could add the logic, for standalone template only, to set the number of replica to 1 when .Values.replicaCount >1 and to set the replicaCount= .Values.replicaCount when it is <= 1. 

What are your thoughts ?

**Comment by FraPazGal** [comment]
Yes, I think that could work @rubij. We'll also need to clearly state this logic in the `values.yaml` to avoid as much confusion as possible.

**Comment by rubij** [comment]
> Yes, I think that could work @rubij. We'll also need to clearly state this logic in the `values.yaml` to avoid as much confusion as possible.

@FraPazGal I have updated the PR, could you review and approve it if my changes are ok for you ?

**Review by FraPazGal — COMMENTED** [review]
The changes LGTM overall @rubij, I just added a minor comment as final review

File: bitnami/mongodb/values.yaml [hunk 1]
```diff
@@ -445,8 +445,9 @@ annotations: {}
 ## @param labels Annotations to be added to the MongoDB(&reg;) statefulset. Evaluated as a template
 ##
 labels: {}
-## @param replicaCount Number of MongoDB(&reg;) nodes (only when `architecture=replicaset`)
-## Ignored when mongodb.architecture=standalone
+## @param replicaCount Number of MongoDB(&reg;) nodes
+## When `mongodb.architecture=replicaset`, the number of replicas is taken in account
+## When `mongodb.architecture=standalone`, the number of replicas can only be 0 or 1 (value higher then 1 will not be taken in account)
```
[the remark this answers is not in the document]
Comment (rubij) [reply bitnami/mongodb/values.yaml:450]: @FraPazGal for me it look all good, could you merge the PR ?

**Comment by rubij** [comment]
@FraPazGal could you merge the Pull Request ?

**Review by FraPazGal — APPROVED** [review]

**Pull request merged**